src/Stream.re

```diff
@@ -1,190 +1,382 @@
+      ) =>
+      'ws =
+      "off";
+    [@bs.send]
+    external onDrainOnce:
+      (
+        subtype([> writable('w)]) as 'ws,
+        [@bs.as "drain"] _,
+        [@bs.uncurry] (unit => unit)
+      ) =>
+      'ws =
+      "once";
+    [@bs.send]
+    external onFinishOnce:
+      (
+        subtype([> writable('w)]) as 'ws,
+        [@bs.as "finish"] _,
+        [@bs.uncurry] (unit => unit)
+      ) =>
+      'ws =
+      "once";
+    [@bs.send]
+    external onPipeOnce:
+      (
+        subtype([> writable('w)]) as 'ws,
+        [@bs.as "pipe"] _,
+        [@bs.uncurry] (subtype([> readable('r)]) => unit)
+      ) =>
+      'ws =
+      "once";
+    [@bs.send]
+    external onUnpipeOnce:
+      (
+        subtype([> writable('w)]) as 'ws,
+        [@bs.as "unpipe"] _,
+        [@bs.uncurry] (subtype([> readable('r)]) => unit)
+      ) =>
+      'ws =
+      "once";
+  };
+  module Impl = {
+    include Common.Impl;
+    include Events;
+    [@bs.send] external cork: subtype([> writable('w)]) => unit = "cork";
+    [@bs.send] external end_: subtype([> writable('w)]) => unit = "end";
+    [@bs.send] external uncork: subtype([> writable('w)]) => unit = "uncork";
+    [@bs.send]
+    external write: (subtype([> writable('w)]), 'w) => bool = "write";
+    [@bs.send]
+    external writeWith:
+      (
+        subtype([> writable('w)]),
+        'w,
+        ~callback: Js.Nullable.t(Js.Exn.t) => unit=?,
+        unit
+      ) =>
+      bool =
+      "write";
+    [@bs.get]
+    external writable: subtype([> writable('w)]) => bool = "writable";
     [@bs.get]
-    external destroyed: subtype('w, 'r, [> kind]) => bool = "destroyed";
+    external writableEnded: subtype([> writable('w)]) => bool =
+      "writableEnded";
+    [@bs.get]
+    external writableCorked: subtype([> writable('w)]) => bool =
+      "writableCorked";
+    [@bs.get]
+    external writableFinished: subtype([> writable('w)]) => bool =
+      "writableFinished";
+    [@bs.get]
+    external writableLength: subtype([> writable('w)]) => int =
+      "writableLength";
+    [@bs.get]
+    external writableHighWaterMark: subtype([> writable('w)]) => int =
+      "writableHighWaterMark";
+    [@bs.get]
+    external writableObjectMode: subtype([> writable('w)]) => bool =
+      "writableObjectMode";
   };
   include Impl;
+
+  type nonrec t('w) = subtype(writable('w));
+  type objStream('w) = subtype([ writable('w) | objectMode]);
+  type supertype('w, 'ty) = subtype([< writable('w)] as 'ty);
+  type nonrec subtype('w, 'ty) = subtype([> writable('w)] as 'ty);
+
+  type makeOptions('w);
+  [@bs.obj]
+  external makeOptions:
+    (
+      ~highWaterMark: int=?,
+      ~objectMode: [@bs.as {json|false|json}] _,
+      ~emitClose: bool=?,
+      ~autoDestroy: bool=?,
+      ~destroy: [@bs.this] (
+                  (
+                    t('w),
+                    ~error: Js.nullable(Js.Exn.t),
+                    ~callback: (~error: option(Js.Exn.t)) => destroy_done
```

Comment:
Added labels to method arguments to help users understand the complex signatures.